kuma/users/views.py

```diff
@@ -685,9 +685,17 @@ def get_form(self, form_class=None):
 
         # This magic sauce only exists whilst this Django view is being used
         # both by Yari (in testing) and by old Kuma (production).
-        # Once Kuma id divorced from doing any
+        # Once Kuma is divorced from doing any HTML rendering, we can refactor this.
         is_yari_signup = self.request.session.get("yari_signup", False)
         if is_yari_signup:
+
+            if not email:
+                for data in extra_email_addresses:
+                    if data["primary"] or not email:
+                        email = data["email"]
+                        break
```

Comment:
This confuses me. As written, `if data["primary"] or not email` is effectively the same as `if True` since `not email` will always be `True`. Did you want to prioritize the primary email, otherwise take the first one? If so, I suspect this should be:
```suggestion
                    if data["primary"] or not email:
                        email = data["email"]
                        if data["primary"]:
                            break
```

Reply:
What about just 
```suggestion
                    email = data["email"]
                    if data["primary"]:
                        break
```

Reply:
I like it! 👍

Reply:
It'll always take the last email though if there isn't a primary, but I suppose we don't care?

Reply:
No, we do care. Let's try this again.
The spec has to be this, right?

1. Pick the *first* `primary` email
2. If none of them are `primary`, pick the first one

Reply:
It's awfully verbose now but the business logic is correct.